pom.xml

```diff
@@ -84,20 +85,27 @@
         <basepom.test.reuse-vm>true</basepom.test.reuse-vm>
         <basepom.test.timeout>240</basepom.test.timeout>
 
+        <dep.annotation-api.version>1.3.1</dep.annotation-api.version>
         <dep.antlr.version>4.7.2</dep.antlr.version>
         <dep.checkstyle.version>8.16</dep.checkstyle.version>
         <dep.dokka.version>0.9.17</dep.dokka.version>
+        <dep.h2.version>1.3.173</dep.h2.version>
         <dep.immutables.version>2.7.1</dep.immutables.version>
         <dep.jackson2.version>2.10.1</dep.jackson2.version>
         <dep.jetbrainsAnnotations.version>13.0</dep.jetbrainsAnnotations.version>
+        <dep.junit5.version>5.5.2</dep.junit5.version>
         <dep.kotlin.version>1.3.61</dep.kotlin.version>
         <dep.plugin.checkstyle.version>3.0.0</dep.plugin.checkstyle.version>
         <dep.plugin.pmd.version>3.9.0</dep.plugin.pmd.version>
         <dep.plugin.sortpom.version>2.8.0</dep.plugin.sortpom.version>
         <dep.plugin.spotbugs.version>3.1.10</dep.plugin.spotbugs.version>
         <dep.pmd.version>6.14.0</dep.pmd.version>
+        <dep.r2dbc.version>Arabba-SR2</dep.r2dbc.version>
+        <dep.reactor-netty.version>0.9.4.RELEASE</dep.reactor-netty.version>
```

Comment:
Did you consider [SmallRye Mutiny](https://smallrye.io/smallrye-mutiny/index.html) also?

At this stage (in my opinion) it's comparable with Reactor and/or RxJava — plus it's already part of MicroProfile.